Buildscript updates

modmuss50 opened the pull request:
* Upgrading to loom 0.6 and the new run config system
* Fix some of the gradle 7 deprecation warnings, there are more left in loom. Once complelte I want to fail the build on deprecations.
* Start looking at singing the jars - currently just pgp. jar singing is TODO
* Build reproducable jars
* Build testmod jars - a fat jar depends on some loom work

modmuss50 committed:
Start upgrading to loom 0.6
Fix some of the gradle 7 deprecation warnings, there are more left in loom.
Start looking at singing the jars

modmuss50 commented:
This PR is currenly locked behind some loom issues.

i509VCB commented on build.gradle:
```diff
@@ -112,18 +114,25 @@ allprojects {
+		runs {
+			testmodClient {
+				client()
+				ideConfigGenerated false
```
Why not generate an ide run config?

modmuss50 replied:
Becuase you would have over 100 of them, should just do that for the root project thinking about it.

modmuss50 replied:
Done.

modmuss50 committed:
Generate testmod run config on the root project.

liach commented:
what's the lowest gradle supported after this update? imo feel free to bump

modmuss50 commented:
Whatever the wrapper is set to is the only "supported" version. I dont see why you would want to try and use it on anything else. This does include most of the required changes for 7 tho.

liach commented:
Can we use a gradle exec to call [jdk's jarsigner](https://docs.oracle.com/en/java/javase/15/docs/specs/man/jarsigner.html) to sign jars? A side note: If we do so, it should in an extra `doLast` for a jar task since this tool seems to change jar files in-place

modmuss50 commented:
No, the jars are signed on a remote server, my plugin does not change the jars in place.

liach commented:
After looking at your [remote sign](https://github.com/modmuss50/RemoteSign/blob/master/src/main/java/me/modmuss50/remotesign/RemoteSignExtension.java) I have a few comments:
1. The 3 string fields should probably be made task-specific. Can keep it in extension so task can query from extension in case the string properties aren't set.
2. Consider merging https://github.com/modmuss50/RemoteSign/blob/e6913049148a71cb9c0c66d2c069625d4f57271d/src/main/java/me/modmuss50/remotesign/RemoteSignJarTask.java#L35-L57 and https://github.com/modmuss50/RemoteSign/blob/e6913049148a71cb9c0c66d2c069625d4f57271d/src/main/java/me/modmuss50/remotesign/RemoteSignExtension.java#L92-L114 to one common utility method (maybe merge the requirements for file, requestUrl, pgpKey to that new method as well)

Technici4n committed:
Fabric API Lookup (#1234)

# Fabric API Lookup API v1
## Introduction
This module allows Api instances to be associated with game objects without specifying how the association is implemented. This is useful when the same Api could be implemented more than once or implemented in different ways.

Many thanks to @Grondag for providing the original concept (#1072).
Thanks also go to @i509VCB, @Pyrofab, @sfPlayer1 and the others who were involved with the design of this module.

This is the foundation upon which can be built for example a fluid transfer api (#1166). Closes #1199.

## Flexible Api Querying
## Block Api Usage example
## Building blocks
This PR was changed a lot, please have a look at the README, the package info, and the javadoc for `BlockApiLookup` and `ApiLookupMap` for up-to-date documentation.

## More usage examples
FastTransferLib (https://github.com/Technici4n/FastTransferLib) is an experiment to build an item, fluid and energy transfer api on top of this module. (Which was until recently called `fabric-provider-api-v1`.)

## Missing things?
~~I could add an overload of `BlockApiLookup#find` with nullable `BlockState` and `BlockEntity` parameters, so that the caller can directly provide them if they are available for some reason.~~ Added in later commits.

There is no module to retrieve apis from items or entities yet because there were unsolved issues with those. The community can use the provided building blocks to experiment with their own implementations of `ItemStackApiLookup` and `EntityApiLookup` until the way forward becomes clear, but let's please not delay the `BlockApiLookup` because of that.

Co-authored-by: i509VCB <git@i509.me>
Co-authored-by: PepperBell <44146161+PepperCode1@users.noreply.github.com>

modmuss50 committed:
Bump version.

modmuss50 committed:
Fat testmod jar

jpenilla committed:
Fix sendError discrepency with client commands (#1358)

This fixes a discrepancy between how error messages are formatted in client commands vs server commands.

Regnander committed:
Create sv_se.json (#1388)

* Create sv_se.json

* Update sv_se.json

modmuss50 committed:
Bump version

modmuss50 committed:
Enable parallel by default, update to gradle 7

modmuss50 commented:
Uhh, I broke this fork.
